### PR TITLE
Collection expressions: tests for spreads of dynamic and non-generic IEnumerable collections

### LINKED ISSUE #69704: Collection expressions: conversion error reported for spread of `dynamic` type
**Version Used**:  e85f97f5

**Steps to Reproduce**:

Compile:
```csharp
class Program
{
    static void Main()
    {
        dynamic x = new[] { 2, 3 };
        int[] y = [1, ..x]; // error: cannot convert from 'object' to 'int'
    }
}
```
**Expected Behavior**:
Compiles and runs successfully.

**Actual Behavior**:
```
(6,23): error CS1503: Argument 1: cannot convert from 'object' to 'int'
```

### PR DESCRIPTION
**Pull request by cston**
Closes #69704

### CODE REVIEW

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 2)

**Comment by cston** [comment]
@dotnet/roslyn-compiler for a second review since the PR includes a fix for IOperation, thanks.

File: src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs [hunk 1]
```diff
@@ -1287,7 +1287,9 @@ private ISpreadOperation CreateBoundCollectionExpressionSpreadElement(BoundColle
             SyntaxNode syntax = element.Syntax;
             bool isImplicit = element.WasCompilerGenerated;
             var elementType = element.EnumeratorInfoOpt?.ElementType.GetPublicSymbol();
-            var elementConversion = BoundNode.GetConversion(iteratorItem, element.ElementPlaceholder);
+            var elementConversion = iteratorItem?.HasErrors == true ?
```
Comment (333fred) [inline src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs:1290]: This doesn't feel correct. Whether or not the iterator item has errors seems like it should be irrelevant to whether a conversion is present.

Comment (333fred) [reply src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs:1290]: I also don't see any tests related to this change?

**Review by 333fred — COMMENTED** [review]
Done review pass.

File: src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs (hunk 1 above)
Comment (cston) [reply src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs:1290]: The issue is hit in one of the new tests in the IOperation test run, but I've added a specific test: `IOperation_SpreadElementConversion_Untyped`.

Comment (cston) [reply src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs]: > This doesn't feel correct. Whether or not the iterator item has errors seems like it should be irrelevant to whether a conversion is present.

Good catch, thanks. I've updated the check to look for collection initializers specifically.

Comment (cston) [reply src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs]: I've split out the IOperation fix to https://github.com/dotnet/roslyn/pull/71461. This PR is now limited to tests for https://github.com/dotnet/roslyn/issues/69704.

**Review by 333fred — APPROVED** [review]

**Pull request merged**